cluster-autoscaler/cloudprovider/azure/azure_config.go

```diff
@@ -498,23 +303,25 @@ func (cfg *Config) validate() error {
 		return fmt.Errorf("subscription ID not set")
 	}
 
-	if cfg.UseManagedIdentityExtension {
-		return nil
+	if cfg.UseManagedIdentityExtension && cfg.UseFederatedWorkloadIdentityExtension {
+		return fmt.Errorf("you can not combine both managed identity and workload identity as an authentication mechanism")
 	}
 
-	if cfg.TenantID == "" {
-		return fmt.Errorf("tenant ID not set")
-	}
+	if !cfg.UseManagedIdentityExtension && !cfg.UseFederatedWorkloadIdentityExtension {
+		if cfg.TenantID == "" {
+			return fmt.Errorf("tenant ID not set")
+		}
 
-	switch cfg.AuthMethod {
-	case "", authMethodPrincipal:
-		if cfg.AADClientID == "" {
-			return errors.New("ARM Client ID not set")
+		switch cfg.AuthMethod {
+		case "", authMethodPrincipal:
+			if cfg.AADClientID == "" {
+				return fmt.Errorf("ARM Client ID not set")
+			}
+		case authMethodCLI:
+			// Nothing to check at the moment.
+		default:
+			return fmt.Errorf("unsupported authorization method: %s", cfg.AuthMethod)
 		}
-	case authMethodCLI:
-		// Nothing to check at the moment.
-	default:
-		return fmt.Errorf("unsupported authorization method: %s", cfg.AuthMethod)
 	}
 
 	if cfg.CloudProviderBackoff && cfg.CloudProviderBackoffRetries == 0 {
```

Comment:
This used to be unintentionally(?) neglected when `UseManagedIdentityExtension` is `true`, which should be completely unrelated. Current changes should fix this bug.